### PR TITLE
Add ChangeLogService

### PR DESCRIPTION
**Pull request by kapillamba4**
### Description
Added ChangeLogService

### Screenshots
captured

### CODE REVIEW

File: frontend/src/component/UIFactory.tsx [hunk 1]
```diff
@@ -18,6 +18,7 @@ import { UrlService } from '../service/Url.service';
 import { SearchService } from '../service/Search.service';
 import { SearchBar } from './ui/SearchBar';
 import { ViewChangeLogButton } from './ui/ViewChangeLogButton';
+import { ChangeLogService } from '../service/Updates.service';
```
Comment (byliuyang) [inline frontend/src/component/UIFactory.tsx]:
```suggestion
import { ChangeLogService } from '../service/ChangeLog.service';
```

Comment (byliuyang) [reply frontend/src/component/UIFactory.tsx]: Aha, this should change correspondingly, right?

File: frontend/src/component/pages/Home.tsx [hunk 2]
```diff
@@ -91,6 +93,15 @@ export class Home extends Component<Props, State> {
     });
     this.handleStateChange();
     this.autoFillLongLink();
+
+    const changeLog = await this.props.changeLogService.getChangeLog();
+    this.setState({ changeLog }, async () => {
+      if (!(await this.props.changeLogService.hasUpdates())) {
```
Comment (byliuyang) [inline frontend/src/component/pages/Home.tsx]:
```suggestion
      const hasUpdates = await this.props.changeLogService.hasUpdates();
      if (!hasUpdates) {
```

File: frontend/src/service/ChangeLog.service.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      const changeLog = await this.getChangeLog();
+      const lastSeenTimestamp = await this.getLastSeenChangeLog();
+      if (
+        changeLog &&
+        changeLog[0] &&
+        lastSeenTimestamp < changeLog[0].releasedAt
+      ) {
+        resolve(true);
+      }
+
```
Comment (byliuyang) [inline frontend/src/service/ChangeLog.service.ts]:
```suggestion
if (! changeLog) {
    resolve(false);
    return;
}

if (changeLog.length < 1) {
    resolve(false);
    return;
}

if (! lastSeenAt < changeLog[0].releasedAt) {
    resolve(false);
    return;
}
```

File: frontend/src/service/ChangeLog.service.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import { Update } from '../entity/Update';
+
+export class ChangeLogService {
+  hasUpdates(): Promise<boolean> {
+    return new Promise(async (resolve, reject) => {
+      const changeLog = await this.getChangeLog();
+      const lastSeenTimestamp = await this.getLastSeenChangeLog();
```
Comment (byliuyang) [inline frontend/src/service/ChangeLog.service.ts]:
```suggestion
      const lastSeenAt = await this.getLastSeenChangeLogAt();
```

File: frontend/src/service/ChangeLog.service.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        lastSeenTimestamp < changeLog[0].releasedAt
+      ) {
+        resolve(true);
+      }
+
+      resolve(false);
+    });
+  }
+
+  getLastSeenChangeLog(): Promise<number> {
```
Comment (byliuyang) [inline frontend/src/service/ChangeLog.service.ts]:
```suggestion
  getLastSeenChangeLog(): Promise<Time> {
```

Comment (byliuyang) [reply frontend/src/service/ChangeLog.service.ts]: Let's define a new type call Time to clarify what this is.

File: frontend/src/service/ChangeLog.service.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
+import { Update } from '../entity/Update';
+
+type Time = number;
+
+export class ChangeLogService {
+  hasUpdates(): Promise<boolean> {
+    return new Promise(async (resolve, reject) => {
+      const changeLog = await this.getChangeLog();
+      const lastSeenAt = await this.getLastSeenChangeLog();
+      if (!changeLog) {
```
Comment (byliuyang) [inline frontend/src/service/ChangeLog.service.ts]: Move this before line 9. `getLastSeenChangeLog` is a network request and takes time. If there is no changelog, then there is no point of loading `getLastSeenChangeLog`. This also reduces our server load.

Comment (kapillamba4) [reply frontend/src/service/ChangeLog.service.ts]: @byliuyang, getChangeLog is also a network request. Most probably changeLog will not be empty so i don't think moving [these](https://github.com/short-d/short/pull/560/files#diff-b7dccee6e4430bdac2b4f2d5ce242846R10-R18) lines will not have must intended effect

**Pull request merged**